### DISCUSSION TITLE
Can Stylet incorporate with WinUI 3 ?

### DISCUSSION THREAD

**Discussion by sigmarsson** [canton7/Stylet#167] [Q&A]
Hi,

Is it possible to harness the Model-First approach and all other benefits of Stylet while building a WinUI 3 desktop application ?

**Comment by canton7** [comment]
I'm afraid I haven't looked into WinUI, so I can't say off the top of my head.

**Comment by sigmarsson** [comment]
Okay, maybe I am happy with WinUI 2.4 too. Is it possible to harness Stylet inside a Universal desktop application targeting Windows 10 version 1903 ? I am not sure...

**Comment by canton7** [comment]
Stylet currently only supports WPF. Not UWP or WinUI. It may make sense to port Stylet to WinUI, but this is not something I've looked into.

**Comment by sigmarsson** [comment]
Okay, I received your former message. .NET 5 will be supported therefore we can shun the entire UWP track by opting for WinUI 3. It sounds so splendid, it is gonna be an evolution of UX by watching the latest Windows 10 controls empowered by the ones Syncfusion & Telerik bestow for the community. 👍

**Comment by canton7** [comment]
No, Stylet does not support WinUI, and porting it will take effort. I don't know if or when this will ever happen. 

Stylet will definitely support **WPF** on .NET 5 because Stylet is a WPF framework, but WPF is not WinUI.

**Comment by sigmarsson** [comment]
Okay, no stress I believe it isn't 5-mouse-click job, I remain then a visitor of your work and see whenever you may publish such aspiration to follow that porting through.    
Best of luck & Thank you for supporting on this matter.

**Comment by sigmarsson** [comment]
Perhaps, porting Stylet to WinUI isn't essential since WinUI 3 will support xaml islands hence a WPF desktop modernization is gonna be feasible piece by piece on the user interface. As MS moniker it is the **hybrid use**. The current WinUI 3 preview 3 does not however support the islands. I think next year's WinUI 3 release and an elevation of Stylet target framework to .NET 5 will together unlock many possibilities to enrich an older Win32 desktop application. The OS though must be as up to date as Windows 10 (1803+), which was an April 2018 Update.

**Comment by sigmarsson** [comment]
However, WinUI 3.0 will **not** support xaml islands, only a later version 3.x, which may come in either 2021 or 2022.  [Here is WinUI roadmap](https://github.com/microsoft/microsoft-ui-xaml/blob/master/docs/roadmap.md)

**Comment by canton7** [comment]
If you're writing a new WinUI app I don't think you'd want to write everything as WPF in a XAML island though: if you're doing that, why not just write a WPF app to start with? All you'd be doing is adding complexity and limitations, without gaining any benefits.

**Comment by sigmarsson** [comment]
Yeah, no way. Just thinking whether I should rewrite an application with WinUI 3 and departure from Stylet, **or** wait for Stylet 1.4 or maybe 2.0 that supports a WinUI project. By today I learnt that the solution with the islands is uncertain in time.
